src/input/input_python.rs

```diff
@@ -303,7 +303,9 @@ impl<'a> Input<'a> for PyAny {
             if PyBool::is_exact_type_of(self) {
                 Err(ValError::new(ErrorTypeDefaults::IntType, self))
             } else {
-                Ok(EitherInt::Py(self))
+                // force to an int to upcast to a pure python int
```

Comment:
> we now do the same in "strict" mode to match lax mode

i would definitely consider reverting this change - substituting consistency between lax and strict for a smaller change of breaking changes.